systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessST.java

```diff
@@ -0,0 +1,184 @@
+import io.fabric8.kubernetes.api.model.NodeAffinity;
+import io.fabric8.kubernetes.api.model.NodeSelectorRequirement;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodAffinityTerm;
+import io.strimzi.api.kafka.model.KafkaConnectResources;
+import io.strimzi.api.kafka.model.KafkaMirrorMaker2Resources;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.api.kafka.model.Rack;
+import io.strimzi.systemtest.AbstractST;
+import static io.strimzi.systemtest.Constants.CONNECT;
+import static io.strimzi.systemtest.Constants.MIRROR_MAKER2;
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import io.strimzi.systemtest.annotations.ParallelNamespaceTest;
+import io.strimzi.systemtest.annotations.ParallelSuite;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import io.strimzi.systemtest.resources.crd.KafkaConnectResource;
+import io.strimzi.systemtest.storage.TestStorage;
+import io.strimzi.systemtest.templates.crd.KafkaConnectTemplates;
+import io.strimzi.systemtest.templates.crd.KafkaMirrorMaker2Templates;
+import io.strimzi.systemtest.templates.crd.KafkaTemplates;
+import io.strimzi.systemtest.utils.StUtils;
+import io.strimzi.systemtest.utils.kafkaUtils.KafkaConnectUtils;
+import io.strimzi.systemtest.utils.kubeUtils.objects.PodUtils;
+import static io.strimzi.test.k8s.KubeClusterResource.cmdKubeClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+@Tag(REGRESSION)
+@ParallelSuite
+class RackAwarenessST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(RackAwarenessST.class);
+    private static final String TOPOLOGY_KEY = "kubernetes.io/hostname";
+
+    @ParallelNamespaceTest
+    void testKafkaRackAwareness(ExtensionContext extensionContext) {
+        TestStorage storage = storageMap.get(extensionContext);
+        String namespace = storage.getNamespaceName();
+        String clusterName = storage.getClusterName();
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaEphemeral(clusterName, 1, 1)
+                .editSpec()
+                    .editKafka()
+                        .withNewRack(TOPOLOGY_KEY)
+                        .addToConfig("replica.selector.class", "org.apache.kafka.common.replica.RackAwareReplicaSelector")
+                    .endKafka()
+                .endSpec()
+                .build());
+
+        LOGGER.info("Kafka cluster deployed successfully");
+        String ssName = KafkaResources.kafkaStatefulSetName(clusterName);
+        String podName = PodUtils.getPodNameByPrefix(namespace, ssName);
+        Pod pod = kubeClient().getPod(namespace, podName);
```

Comment:
About this, it would be much clearer to use just `Pod pod = kubeClient().getPod(namespace, KafkaResources.kafkaPodName(testStorage.getClusterName(), 0);` then doing these extra operations.

Reply:
Yeah, that's for consistency with the other tests, which actually requires them to avoid repetition.